packages/field/src/DisabledMixin.js

```diff
@@ -0,0 +1,118 @@
+import { dedupeMixin } from '@lion/core';
+
+/**
+ * #DisabledMixin
+ *
+ * For now this is a private mixin for option
+ * Do not use it anywhere else - it will become part of field eventually.
+ *
+ * @polymerMixin
+ * @mixinFunction
+ */
+export const DisabledMixin = dedupeMixin(
+  superclass =>
+    // eslint-disable-next-line no-shadow
+    class DisabledMixin extends superclass {
+      static get properties() {
+        return {
+          disabled: {
+            type: Boolean,
+            reflect: true,
+          },
+          // we us a property here as if we use the native tabIndex we can not set a default value
+          // in the constructor as it synchronously sets the attribute which is not allowed in the
+          // constructor phase
+          tabIndex: {
+            type: Number,
+            reflect: true,
+            attribute: 'tabindex',
+          },
+        };
+      }
+
+      static get disableMixinManagesTabIndex() {
+        return true;
+      }
+
+      constructor() {
+        super();
+        this.__requestedToBeDisabled = false;
+        this.__isUserSettingDisabled = true;
+        this.__isUserSettingTabIndex = true;
+
+        this.__restoreDisabledTo = false;
+        this.__restoreTabIndexTo = 0;
+        this.disabled = false;
+        if (this.constructor.disableMixinManagesTabIndex) {
+          this.__internalSetTabIndex(0);
+        }
+      }
+
+      makeRequestToBeDisable() {
+        this.__requestedToBeDisabled = true;
+        this.__restoreDisabledTo = this.disabled;
+        this.__restoreTabIndexTo = this.tabIndex;
+        this.__internalSetDisabled(true);
+      }
```

Comment:
looks like a bug to me, because if `makeRequestToBeDisable` is called 2 times in a row, you will loose the original state

maybe just add a guard `if (this.__requestedToBeDisabled) { return; }` im the beginning of the method

maybe start with a test for this case :)

Reply:
added test which showed that a guard was needed 👍